sdk/core/core-rest-pipeline/src/policies/armChallengeAuthenticationPolicy.ts

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { GetTokenOptions } from "../../../core-auth/types/latest/core-auth";
+import { PipelinePolicy } from "../pipeline";
+import {
+  AuthorizeRequestOnChallengeOptions,
+  bearerTokenAuthenticationPolicy,
+  BearerTokenAuthenticationPolicyOptions,
+  defaultAuthorizeRequest
+} from "./bearerTokenAuthenticationPolicy";
+
+/**
+ * The programmatic identifier of the bearerTokenAuthenticationPolicy.
+ */
+export const ARMChallengeAuthenticationPolicyName = "armChallengeAuthenticationPolicy";
+
+/**
+ * Options to configure the armChallengeAuthenticationPolicy
+ */
+export interface ARMChallengeAuthenticationPolicyOptions
+  extends BearerTokenAuthenticationPolicyOptions {}
+
+/**
+ * Converts a uint8Array to a string.
+ * @internal
+ */
+export function uint8ArrayToString(ab: Uint8Array): string {
+  const decoder = new TextDecoder("utf-8");
```

Comment:
`TextDecoder` only became global in node 11, and aren't we still supporting node10?